bin/lib/zopen-build

```diff
@@ -343,33 +367,73 @@ checkEnv()
-  fi
-
-  if [ "${ZOPEN_TYPE}x" = "TARBALLx" ]; then
-    if [ "${ZOPEN_TARBALL_URL}x" = "x" ]; then
-      export ZOPEN_TARBALL_URL="${ZOPEN_URL}"
-    fi
-    if [ "${ZOPEN_TARBALL_DEPS}x" = "x" ]; then
-      export ZOPEN_TARBALL_DEPS="${ZOPEN_DEPS}"
-    fi
-    printHeader "Using tarball environment"
-  elif [ "${ZOPEN_TYPE}x" = "GITx" ]; then
-    if [ "${ZOPEN_GIT_URL}x" = "x" ]; then
-      export ZOPEN_GIT_URL="${ZOPEN_URL}"
-    fi
-
-    if [ "${ZOPEN_GIT_DEPS}x" = "x" ]; then
-      export ZOPEN_GIT_DEPS="${ZOPEN_DEPS}"
+  if [ ! -z "$ZOPEN_BUILD_LEVEL" ]; then
+    if [ "${ZOPEN_BUILD_LEVEL}x" = "DEVx" ]; then
+      if [ -z "$ZOPEN_DEV_DEPS" ]; then
+        printError "Building from dev, but ZOPEN_DEV_DEPS not specified"
+      fi
+      ZOPEN_DEPS="$ZOPEN_DEV_DEPS"  
+      if [ -z "$ZOPEN_DEV_URL" ]; then
+        printError "Building from dev, but ZOPEN_DEV_URL not specified"
+      fi
+      ZOPEN_URL="$ZOPEN_DEV_URL"  
+      if [ ! -z "$ZOPEN_DEV_BRANCH" ]; then
+        export ZOPEN_GIT_BRANCH="$ZOPEN_DEV_BRANCH"
+      fi
+      if [ ! -z "$ZOPEN_DEV_TAG" ]; then
+        export ZOPEN_GIT_TAG="$ZOPEN_DEV_TAG"
+      fi
+    elif [ "${ZOPEN_BUILD_LEVEL}x" = "STABLEx" ]; then
+      if [ -z "$ZOPEN_STABLE_DEPS" ]; then
+        printError "Building from stable, but ZOPEN_STABLE_DEPS not specified"
+      fi
+      ZOPEN_DEPS="$ZOPEN_STABLE_DEPS"  
+      if [ -z "$ZOPEN_STABLE_URL" ]; then
+        printError "Building from stable, but ZOPEN_STABLE_URL not specified"
+      fi
+      ZOPEN_URL="$ZOPEN_STABLE_URL"  
+      if [ ! -z "$ZOPEN_STABLE_BRANCH" ]; then
+        export ZOPEN_GIT_BRANCH="$ZOPEN_STABLE_BRANCH"
+      fi
+      if [ ! -z "$ZOPEN_STABLE_TAG" ]; then
+        export ZOPEN_GIT_TAG="$ZOPEN_STABLE_TAG"
+      fi
+    else
+      printError "Unrecognized build level ${ZOPEN_BUILD_LEVEL}"
+    fi 
+    extension="${ZOPEN_URL##*.}"
+    if [ "$extension" = "git" ]; then
+      ZOPEN_TYPE="GIT"
+    else
+      ZOPEN_TYPE="TARBALL"
     fi
-    printHeader "Using git environment"
-  elif [ "${ZOPEN_TYPE}x" = "BAREx" ]; then
-    printHeader "Using bare environment"
   else
-    printError "ZOPEN_TYPE must be one of TARBALL or GIT. ZOPEN_TYPE=${ZOPEN_TYPE} was specified"
+    if [ "${ZOPEN_TYPE}x" = "TARBALLx" ]; then
+      if [ ! -z "${ZOPEN_TARBALL_URL}" ]; then
+        export ZOPEN_URL="${ZOPEN_TARBALL_URL}"
+      fi
+      if [ ! -z "${ZOPEN_TARBALL_DEPS}" ]; then
+        export ZOPEN_DEPS="${ZOPEN_TARBALL_DEPS}"
+      fi
+      printHeader "Using tarball environment"
+    elif [ "${ZOPEN_TYPE}x" = "GITx" ]; then
+      if [ ! -z "${ZOPEN_GIT_URL}" ]; then
+        export ZOPEN_URL="${ZOPEN_GIT_URL}"
+      fi
+      if [ ! -z "${ZOPEN_GIT_DEPS}" ]; then
+        export ZOPEN_DEPS="${ZOPEN_GIT_DEPS}"
+      fi
+      printHeader "Using git environment"
+    elif [ "${ZOPEN_TYPE}x" = "BAREx" ]; then
+      printHeader "Using bare environment"
+    else
+      extension="${ZOPEN_URL##*.}"
```

Comment:
isn't this block of code redundant?

Reply:
This is to cover the case where ZOPEN_TYPE is not specified and implicit, we can remove it at some point since that path is deprecated.